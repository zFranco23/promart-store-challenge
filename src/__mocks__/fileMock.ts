export const mockProducts = [
  {
    id: 1,
    title: 'title1',
    price: 5,
    category: 'c1',
    description: 'd1',
    image: 'image',
    rating: {
      count: 5,
      rate: 10,
    },
  },
  {
    id: 2,
    title: 'title2',
    price: 10,
    category: 'c2',
    description: 'd2',
    image: 'image',
    rating: {
      count: 6,
      rate: 19,
    },
  },
];
