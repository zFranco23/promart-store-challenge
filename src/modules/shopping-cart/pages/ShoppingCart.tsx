import { useNavigate } from 'react-router-dom';
import MainButton from '../../../common/components/MainButton/MainButton';
import { FromMobileElem, MobileElem } from '../../../utils/responsive';
import useShoppingCart from '../hooks/useShoppingCart';
import ItemsContent from '../components/ItemsContent/ItemsContent';

const ShoppingCart = () => {
  const navigate = useNavigate();

  const { items, totalPrice } = useShoppingCart();

  const handleGoBack = () => {
    navigate(-1);
  };

  const hastCartItems = items.length > 0;

  const handleCheckout = () => {
    if (hastCartItems) console.log('Checkout process');
    else navigate('/');
  };

  return (
    <div>
      <MobileElem>
        <div className='h-screen v-screen'>
          <div className='h-full flex flex-col justify-between p-8'>
            <div className='flex relative items-center'>
              <div className='absolute left-0 top-0.5'>
                <button onClick={handleGoBack}>
                  <i className='material-icons text-orange'>arrow_back</i>
                </button>
              </div>

              <p className='text-center m-auto font-bold text-orange'>Mi carrito</p>
            </div>
            {hastCartItems && (
              <p className='font-bold mt-8'>
                ¡Tienes {items.length} producto{items.length > 1 ? 's' : ''} en tu carrito!
              </p>
            )}
            <ItemsContent hasCartItems={hastCartItems} items={items} />

            <div>
              {hastCartItems && (
                <div className='flex justify-between mb-4'>
                  <p>Total</p>
                  <p className='text-orange font-bold'>{totalPrice}</p>
                </div>
              )}
              <MainButton className='w-full' onClick={handleCheckout}>
                {hastCartItems ? '¡Lo quiero!' : 'Ver productos'}
              </MainButton>
            </div>
          </div>
        </div>
      </MobileElem>
      <FromMobileElem>wwwwq</FromMobileElem>
    </div>
  );
};

export default ShoppingCart;
