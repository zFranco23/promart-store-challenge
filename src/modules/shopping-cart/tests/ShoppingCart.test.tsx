import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { shallow } from 'enzyme';
import { mockProducts } from '../../../__mocks__/fileMock';
// import { ShoppingCartItem } from '../../../entities';

import ShoppingCart from '../pages/ShoppingCart';
import { RootState } from '../..';

const mockStore = configureStore();

describe('Shopping cart module test', () => {
  let store: ReturnType<typeof mockStore>;
  let wrapper: React.ReactElement;
  beforeEach(() => {
    store = mockStore({
      checkout: {},
      products: {},
      auth: {},
      shoppingCart: {
        items: [mockProducts[0]],
      },
    });
    wrapper = (
      <Provider store={store}>
        <ShoppingCart />
      </Provider>
    );
  });

  test('Shopping cart page should render correctly', () => {
    expect(shallow(wrapper)).toMatchSnapshot();
  });

  test('Should add item to cart by ', () => {
    const {
      shoppingCart: { items },
    } = store.getState() as RootState;

    expect(shallow(wrapper)).toMatchSnapshot();
    expect(items.length).toBe(1);
  });
});
