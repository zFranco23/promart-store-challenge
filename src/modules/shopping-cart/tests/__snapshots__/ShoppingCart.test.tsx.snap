// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shopping cart module test Shopping cart page should render correctly 1`] = `
<ContextProvider
  value={
    {
      "getServerState": undefined,
      "store": {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ShoppingCart />
</ContextProvider>
`;

exports[`Shopping cart module test Should add item to cart by  1`] = `
<ContextProvider
  value={
    {
      "getServerState": undefined,
      "store": {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ShoppingCart />
</ContextProvider>
`;
